src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -545,4 +546,8 @@ await _repositoryCloneManager.PrepareCloneAsync(
 
     protected override NativePath GetEngCommonPath(NativePath sourceRepo) => sourceRepo / VmrInfo.SourceDirName / "arcade" / Constants.CommonScriptFilesPath;
     protected override bool TargetRepoIsVmr() => false;
+    // During backflow, we're flowing a specific VMR commit that the build was built from, so we should just check it out
+    protected override bool ShouldResetBranchToRemoteWhenPreparingVmr() => false;
+    // When flowing local repos, we should never reset branches to the remote ones, we might lose some changes devs wanted
+    protected override bool ShouldResetBranchToRemoteWhenPreparingRepo() => false;
```

Comment:
I am thinking whether this class should not go abstract and we would add `DarcVmrBackflower` as a contrary to `PcsVmrBackflower` so that we can implement the differences more visibly and not just override the Darc behaviour with the PCS one.

Reply:
I think that would make things more clear and easy to read, but maybe let's do that in a follow up issue?

Reply:
Agreed

Reply:
https://github.com/dotnet/arcade-services/issues/4351